### PR TITLE
merging finished todo ( todo-list.component.html branch) to main branch

### PR DESCRIPTION
**Pull request by AndrewLam288**
merging finished todo ( todo-list.component.html branch) to main branch

### CODE REVIEW

**Review by AndrewLam288 — COMMENTED** [review]
ng e2e and ng tests, gradle build work. 
the todos filter parameters also work (including combination).
the todo page look pretty friendly to use for me.


Only waiting for approval to merge, no merge conflict

**Review by TyBeas — APPROVED** [review]

**Pull request merged**
